frontend/static/js/services/tournamentService.js

```diff
@@ -1,33 +1,58 @@
-import ARequestService from "./ARequestService.js";
-import backendURL from "../constants.js";
-import NotImplementedError from "../exceptions/NotImplementedException.js";
+import ARequestService from './ARequestService.js';
+import backendURL from '../constants.js';
+import NotImplementedError from '../exceptions/NotImplementedException.js';
 
-export default class TournamentService extends ARequestService{
+export default class TournamentService extends ARequestService {
 	constructor() {
 		super();
 	}
 
 	async getAll() {
-		throw new NotImplementedError("GET (resource) not implemented for TournamentService");
+		throw new NotImplementedError('GET (resource) not implemented for TournamentService');
 	}
 
 	async getRequest(id) {
-		let url = `${backendURL.tournamentURL}/` + (id ? id : "");
-		const response = await super.getRequest(url);
+		let url = `${backendURL.tournamentURL}/` + (id ? id : '');
+		const response = super.getRequest(url);
 		return response;
 	}
 
+	async getNextMatch(context) {
+		return super.getRequest(`${backendURL.tournamentURL}${context.tournament_id}/matches/${context.match_id}`);
+	}
+
 	async postRequest(data) {
-		const response = await super.postRequest(backendURL.tournamentURL, data);
+		const response = super.postRequest(
+			backendURL.tournamentURL,
+			JSON.stringify({
+				name: data.tournamentName,
+				player_amount: data.numberOfPlayers,
+				host_user_display_name: '',
+			})
+		);
 		return response;
 	}
 
-	async putRequest(data) {
-		const response = await super.putRequest(backendURL.tournamentURL, data);
+	async postPlayer(tournamentData, context) {
+		const response = super.postRequest(
+			`${backendURL.tournamentURL}${context.tournamentId}/players/`,
+			JSON.stringify({
+				username: tournamentData.username,
+				password: tournamentData.password,
+			})
+		);
 		return response;
 	}
 
-	async deleteRequest(){
-		throw new NotImplementedError("DELETE not implemented for TournamentService");
+	async patchRequest(context) {
+		return super.patchRequest(backendURL.tournamentURL, context.tournamentId);
```

Comment:
mmeh, the patch body is supposed to look like:
{
  'op' : 'replace',
  'path' : 'tournament.status',
  'value': 'lobby'
}

to actually be a valid expected patch call, but we dont have time to do the right thing..